src/qml/RelationCombobox.qml

```diff
@@ -22,22 +26,240 @@ Item {
+
+    onOpened: {
+      searchField.forceActiveFocus()
+    }
+
+    onClosed: {
+      searchField.text = ''
+      searchFeaturePopup.cancel()
+      searchFeaturePopup.finished()
+    }
+
+    Page {
+      anchors.fill: parent
+
+      header: PageHeader {
+        title: fieldLabel
+        showApplyButton: true
+        showCancelButton: true
+        onApply: {
+          searchFeaturePopup.close()
+          searchFeaturePopup.apply()
+          searchFeaturePopup.finished()
+        }
+        onCancel: {
+          searchFeaturePopup.close()
+          searchFeaturePopup.cancel()
+          searchFeaturePopup.finished()
+        }
+      }
+
+      TextField {
+        z: 1
+        id: searchField
+        anchors.left: parent.left
+        anchors.right: parent.right
+
+        placeholderText: qsTr("Search…")
+        placeholderTextColor: Theme.mainColor
+
+        height: fontMetrics.height * 2.5
+        padding: 24
+        bottomPadding: 9
+        font: Theme.defaultFont
+        selectByMouse: true
+        verticalAlignment: TextInput.AlignVCenter
+
+        inputMethodHints: Qt.ImhNoPredictiveText | Qt.ImhNoAutoUppercase
+
+        onDisplayTextChanged: {
+          featureListModel.searchTerm = searchField.displayText
+        }
+      }
+
+      Image {
+        id: clearButton
+        z: 1
+        width: fontMetrics.height
+        height: fontMetrics.height
+        source: Theme.getThemeIcon("ic_clear_black_18dp")
+        sourceSize.width: 20 * screen.devicePixelRatio
+        sourceSize.height: 20 * screen.devicePixelRatio
+        fillMode: Image.PreserveAspectFit
+        anchors.top: searchField.top
+        anchors.right: searchField.right
+        anchors.topMargin: height - 7
+        anchors.rightMargin: height - 7
+        opacity: searchField.text.length > 0 ? 1 : 0.25
+
+        MouseArea {
+          anchors.fill: parent
+          onClicked: {
+            searchField.text = '';
+          }
+        }
+      }
+
+      ScrollView {
+        anchors.left: parent.left
+        anchors.right: parent.right
+        anchors.top: searchField.bottom
+
+        padding: 0
+        ScrollBar.horizontal.policy: ScrollBar.AlwaysOff
+        ScrollBar.vertical.policy: ScrollBar.AsNeeded
+        contentItem: resultsList
+        contentWidth: resultsList.width
+        contentHeight: resultsList.height
+        clip: true
+
+        ListView {
+          id: resultsList
+          anchors.top: parent.top
+          model: featureListModel
+          width: parent.width
+          height: searchFeaturePopup.height - searchField.height - 50
+          clip: true
+
+          delegate: Rectangle {
+            id: delegateRect
+
+            property int idx: index
+
+            anchors.margins: 10
+            height: radioButton.visible ? radioButton.height : checkBoxButton.height
+            width: parent ? parent.width : undefined
+            color: model.checked ? Theme.mainColor : 'transparent'
```

Comment:
Somehow I can't follow here, is this using a `checked` property of the `model`? I would guess from the current item/index? Or is this how the delegate works?
Probably my rusty qml knowledge.

Reply:
It uses the "checked" role name from the current delegate model:
https://github.com/opengisch/QField/blob/master/src/core/featurechecklistmodel.cpp#L60